fish.go

```diff
@@ -158,20 +159,17 @@ func (a *App) prepareFishFlags(flags []Flag, previousCommands []string) []string
 }
 
 func fishAddFileFlag(flag Flag, completion *strings.Builder) {
-	switch f := flag.(type) {
-	case *GenericFlag:
-		if f.TakesFile {
-			return
-		}
-	case *StringFlag:
-		if f.TakesFile {
-			return
-		}
-	case *StringSliceFlag:
-		if f.TakesFile {
+	val := reflect.ValueOf(flag)
+	// if flag is a non-nil pointer to a struct...
```

Comment:
Nit: I'm not a huge fan of inline comments that explain what each line of code is doing.

If it's a sufficiently complex line to warrant a comment, it's sufficiently complex to warrant extracting into a function with a descriptive name. I don't think any line in this function meets that bar.